src/io/src/DataLoaderBridge.cpp

```diff
@@ -0,0 +1,293 @@
+    std::sort(clouds.begin(), clouds.end());
+    data_size = clouds.size();
+    haveCloud = (data_size > 0);
+  }
+
+  boost::optional< const boost::property_tree::ptree& > foundRGB;
+  foundRGB = pt.get_child_optional( "data_path.path_to_rgb" );
+  if(foundRGB)
+  {
+    this->path_to_rgb = pt.get<std::string>("data_path.path_to_rgb");
+    std::string rgb_extension = pt.get<std::string>("data_path.rgb_extension");
+    success = getListFile(path_to_rgb, images, rgb_extension, isRGBFile);
+    std::sort(images.begin(), images.end());
+    data_size = images.size();
+    haveRGB = (data_size > 0);
+  }
+
+  boost::optional< const boost::property_tree::ptree& > foundDepth;
+  foundDepth = pt.get_child_optional( "data_path.path_to_depth" );
+  if(foundDepth)
+  {
+    this->path_to_depth = pt.get<std::string>("data_path.path_to_depth");
+    std::string depth_extension = pt.get<std::string>("data_path.depth_extension");
+    success = getListFile(path_to_depth, depths, depth_extension, isDepthFile);
+    std::sort(depths.begin(), depths.end());
+    data_size = depths.size();
+    haveDepth = (data_size > 0);
+  }
+
+  this->isLoop = pt.get<bool>("option.isLoop", true);
+
+  if(!checkConsistency()){
+    outError("Provided data is not consistent, all must be a file or size of each kind of data must be equal!");
+    success = false;
+  }
+
+  this->frameRate = pt.get<double>("camera_info.frame_rate", -1);
+  this->cameraInfo.width = pt.get<int>("camera_info.width", 640);
+  this->cameraInfo.height = pt.get<int>("camera_info.height", 480);
+
+  this->cameraInfo.roi.width = this->cameraInfo.width;
+  this->cameraInfo.roi.height = this->cameraInfo.height;
+  this->cameraInfo.roi.x_offset = 0;
+  this->cameraInfo.roi.y_offset = 0;
+
+  std::string line = pt.get<std::string>("camera_info.matrix");
+  std::replace(line.begin(), line.end(), ',', ' ');
+
+  std::stringstream sstr(line);
+  std::vector<double> cameraMatrix{std::istream_iterator<double>(sstr),
+                                   std::istream_iterator<double>()};
+  std::copy(cameraMatrix.begin(), cameraMatrix.end(), this->cameraInfo.K.begin());
+
+  line = pt.get<std::string>("camera_info.distortion");
+  std::replace(line.begin(), line.end(), ',', ' ');
+  sstr.str(line);
+  std::vector<double> cameraDistortion{std::istream_iterator<double>(sstr),
+                                       std::istream_iterator<double>()};
+  std::copy(cameraDistortion.begin(), cameraDistortion.end(), this->cameraInfo.D.begin());
+
+  this->depth_scaling_factor = pt.get<int>("camera_info.depth_scaling_factor", 1);
+
+  return success;
+}
+
+bool DataLoaderBridge::setData(uima::CAS &tcas, uint64_t ts){
+  if(!newData())
+    return false;
+
+  outInfo("setData");
+
+  rs::SceneCas cas(tcas);
+
+  if(haveCloud){
+    pcl::PointCloud<pcl::PointXYZRGBA>::Ptr cloud_ptr (new pcl::PointCloud<pcl::PointXYZRGBA>);
+
+    std::string path;
+    if(!isCloudFile){
+      path = clouds[iterator];
```

Comment:
If I set isLoop in the config to false the programm still keeps running and at this point it reads from index 1 when there is only 1 entry. This results in a random string as a path. Not sure if that should be fixed here or if it shouldn't loop at all. Also maybe this should be called index since it's just an integer, iterator is slightly misleading.